_pages/clients/rust.md

```diff
@@ -63,200 +61,200 @@ If you are running [Algorithmia Enterprise](/enterprise), you can specify the AP
-
-	if nlp_directory.exists().unwrap() == false{
-	    nlp_directory.create(DataAcl::default());
-	}
-
-	let text_file = "data://YOUR_USERNAME/nlp_directory/jack_london.txt";
-
-	// Check if file exists
-	if client.file(text_file).exists().unwrap() == false{
-		// Upload local file
-		nlp_directory.put_file("/your_local_path_to_file/jack_london.txt");
-	}
-
-	// Download contents of file as a string
-	if client.file(text_file).exists().unwrap() {
-		let mut text_reader = client.file(text_file).get().unwrap();
-		let mut jack_london_text = String::new();
-		let _ = text_reader.read_to_string(&mut jack_london_text);
-
-		// Call Summarizer algorithm.
-		let algo = client.algo("nlp/Summarizer/0.1.3");
-		match algo.pipe(& jack_london_text) {
-		    Ok(response) => {
-		    	response.as_string().unwrap();
-		    },
-		    Err(err) => println!("error calling Summarizer algo: {}", err),
-		};
-	}
-}
+let download_uri = "data://.algo/dlib/FaceDetection/temp/detected_faces.png";
+{% endhighlight %}
+
+Verify that the file that you want to download exists, and try downloading it to a new local file location. The Rust client allows you to download files by calling `get` on a `DataFile` object, which returns a `Result`-wrapped `DataResponse` that implements `Read`. You can then write the file locally using `std::fs:File` and `std::io:copy`:
+
+{% highlight rust %}
+if client.file(download_uri).exists().unwrap() {
+    let mut png_reader = client.file(download_uri).get().unwrap();
+    let mut png = std::fs::File::create("../data/detected_faces.png").unwrap();
+    std::io::copy(&mut png_reader, &mut png).unwrap();
+};
+{% endhighlight %}
+
+Alternately, if you just need the binary content of the file to be stored in a variable, you could store the results in a byte array:
+
+{% highlight rust %}
+let mut png_reader = client.file(download_uri).get().unwrap();;
+let mut png = Vec::new();
+png_reader.read_to_end(&mut t800_bytes);
 {% endhighlight %}
+
+If the file was text (an image, etc.), you can use `read_to_string` on the response. The [API Specification](/developers/api/#get-a-file-or-directory) has more details on how to get files from a data collection using the Data API.
+
+## Additional Functionality
+
+In addition to the functionality covered in this guide, the Rust Client Library provides a complete interface to the Algorithmia platform, including [managing algorithms](/developers/algorithm-development/algorithm-management), administering [organizations](/developers/platform/organizations), and working with [source control](/developers/algorithm-development/source-code-management). You can also visit the [API Docs](/developers/api) to view the complete API specification.
+
+## Next Steps
+
+If you're a data scientist or developer who will be building and deploying new algorithms, you can move on to the [Algorithm Development > Getting Started](/developers/algorithm-development/your-first-algo/) guide.
```

Comment:
```suggestion
If you're a data scientist or developer who will be building and deploying new algorithms, you can move on to the [Algorithm Development > Getting Started](/developers/algorithm-development/your-first-algo) guide.
```

Reply:
Gotta remember the trailing slashes...

Reply:
lol, we had a huge push a year ago to get that all sorted out. Was several sprints worth of work to optimize across several different apps. It's really not a big deal if one slips through, they're just burned into my brain and they stick out to me now.